nautobot/dcim/models/locations.py

```diff
@@ -114,11 +114,29 @@ def clean(self):
 
 
 class LocationQuerySet(TreeQuerySet):
+    @classmethod
+    def as_manager(cls, with_tree_fields=False):
+        """Get an appropriate TreeManager subclass for working with Locations."""
+        LocationManager = TreeManager.from_queryset(cls)
+        LocationManager._built_with_as_manager = True
+        LocationManager._with_tree_fields = with_tree_fields
+        return LocationManager()
+
     def get_for_model(self, model):
         """Filter locations to only those that can accept the given model class."""
         content_type = ContentType.objects.get_for_model(model._meta.concrete_model)
         return self.filter(location_type__content_types=content_type)
 
+    def natural_key_kwargs(self, *args):
```

Comment:
Now that I have a custom `natural_key_kwargs` on `RestrictedQuerySet`, it feels wrong to also be overriding it in `LocationQuerySet` - need to revisit this.